middleware/error-handler.js

```diff
@@ -1,10 +1,37 @@
-const { CustomAPIError } = require('../errors')
-const { StatusCodes } = require('http-status-codes')
+// const { CustomAPIError } = require('../errors');
+const { StatusCodes } = require('http-status-codes');
 const errorHandlerMiddleware = (err, req, res, next) => {
-  if (err instanceof CustomAPIError) {
-    return res.status(err.statusCode).json({ msg: err.message })
+  let customError = {
+    //set default
+    statusCode: err.statusCode || StatusCodes.INTERNAL_SERVER_ERROR,
+    msg: err.message || 'Something went wrong try again later',
+  };
+
+  // if (err instanceof CustomAPIError) {
+  //   return res.status(err.statusCode).json({ msg: err.message });
+  // }
+
+  if (err.name === 'ValidationError') {
+    customError.msg = Object.values(err.errors)
+      .map((item) => item.message)
+      .join(',');
+    customError.statusCode = 400;
+  }
+
+  if (err.name === 'CastError') {
+    customError.msg = `No item found with id: ${err.value}`;
+    customError.statusCode = 404;
+  }
+  //if there is error object check if that err.code is === 1100
+  if (err.code && err.code === 11000) {
+    //modify values of our customError object
+    customError.msg = `Duplicate value entered for ${Object.keys(
+      err.keyValue
+    )} field, please choose another value`;
+    customError.statusCode = 400;
```

Comment:
Only slight exception to this rule is HTTP status codes, since most web developers will have them memorized, but you also have your `StatusCodes.x` library for the same reason; giving these things names!

You could use a value from that library in this place instead of writing the plain number.